tiledb/sm/rest/test/unit_rest_profile.cc

```diff
@@ -0,0 +1,298 @@
+#include <iostream>
+
+using namespace tiledb::common;
+using namespace tiledb::sm;
+
+std::string cloudtoken_ = "abc123def456";  // Token used by in-test cloud.json.
+
+/* Tracks the expected name and parameter values for a RestProfile. */
+struct expected_values_t {
+  std::string name = RestProfile::DEFAULT_NAME;
+  std::string password = RestProfile::DEFAULT_PASSWORD;
+  std::string payer_namespace = RestProfile::DEFAULT_PAYER_NAMESPACE;
+  std::string token = cloudtoken_;
+  std::string server_address = RestProfile::DEFAULT_SERVER_ADDRESS;
+  std::string username = RestProfile::DEFAULT_USERNAME;
+};
+
+struct RestProfileFx {
+ public:
+  TemporaryLocalDirectory tempdir_;  // A temporary working directory.
+  std::string homedir_;              // The temporary in-test $HOME directory.
+  std::string cloudpath_;            // The in-test path to the cloud.json file.
+
+  RestProfileFx()
+      : tempdir_(TemporaryLocalDirectory("unit_rest_profile"))
+      , homedir_(tempdir_.path())
+      , cloudpath_(homedir_ + ".tiledb/cloud.json") {
+    // Fstream cannot create directories, only files, so create the .tiledb dir.
+    std::filesystem::create_directories(homedir_ + ".tiledb");
+
+    // Write to the cloud path.
+    json j = {
+        {"api_key",
+         {json::object_t::value_type("X-TILEDB-REST-API-KEY", cloudtoken_)}},
+        {"host", RestProfile::DEFAULT_SERVER_ADDRESS},
+        {"username", RestProfile::DEFAULT_USERNAME},
+        {"password", RestProfile::DEFAULT_PASSWORD},
+        {"verify_ssl", "true"}};
+
+    {
+      std::ofstream file(cloudpath_);
+      file << std::setw(2) << j << std::endl;
+    }
+  }
+
+  ~RestProfileFx() = default;
+
+  /** Returns a new RestProfile with the given name, at the in-test $HOME. */
+  RestProfile create_profile(
+      const std::string& name = RestProfile::DEFAULT_NAME) {
+    return RestProfile(name, homedir_);
+  }
+
+  /** Returns true iff the profile's parameter values match the expected. */
+  bool is_valid(RestProfile p, expected_values_t e) {
+    if (p.name() == e.name && p.get_param("rest.password") == e.password &&
+        p.get_param("rest.payer_namespace") == e.payer_namespace &&
+        p.get_param("rest.token") == e.token &&
+        p.get_param("rest.server_address") == e.server_address &&
+        p.get_param("rest.username") == e.username)
+      return true;
+    return false;
+  }
+
+  /**
+   * Returns the RestProfile at the given name from the local file,
+   * as a json object.
+   */
+  json profile_from_file_to_json(std::string filepath, std::string name) {
+    json data;
+    if (std::filesystem::exists(filepath)) {
+      std::ifstream file(filepath);
+      file >> data;
+      file.close();
+      return data[name];
+    } else {
+      return data;
+    }
+  }
+};
+
+TEST_CASE_METHOD(
+    RestProfileFx, "REST Profile: Default profile", "[rest_profile][default]") {
+  // Remove cloud.json to ensure the RestProfile constructor doesn't inherit it.
```

Comment:
NIT: there's one uncovered scenario, if the `.tiledb` directory does not exist. I think it's worth adding a test identical to this but removing the whole directory instead.